File: client/src/service/BannerService.js
import api from "./AxiosConfig";

//메인 배너 가져오기
export const getMainBanners = async () => {
    const res = await api.get("/banner/main")
    return res.data
}

//중간 배너 가져오기
export const getMidBanners = async () => {
    const res = await api.get("/banner/mid")
    return res.data
}

//전체 배너 가져오기 (관리자)
export const getAllBanners = async () => {
    const res = await api.get("/banner/all")
    return res.data
}

//백엔드: POST /api/banner
export const createBanner = async (data) => {
    const res = await api.post("/banner", data)
    return res.data
}

//배너 수정
export const updateBanner = async (id, data) => {
    const res = await api.put(`/banner/${id}`, data)
    return res.data
}

//배너 삭제
export const deleteBanner = async (id) => {
    const res = await api.delete(`/banner/${id}`)
    return res.data
}

File: barofarm/src/main/java/com/barofarm/barofarm/service/BannerService.java
package com.barofarm.barofarm.service;

import java.util.List;
import org.springframework.stereotype.Service;
import lombok.RequiredArgsConstructor;
import com.barofarm.barofarm.repository.BannerRepository;
import com.barofarm.barofarm.entity.Banner;

@Service
@RequiredArgsConstructor
public class BannerService {

    private final BannerRepository bannerRepository;

    // 메인 배너
    public List<Banner> getMainBanners() {
        return bannerRepository.findByUseYnAndPositionOrderBySortOrderAsc("Y", "MAIN");
    }

    // 중간 배너
    public List<Banner> getMidBanners() {
        return bannerRepository.findByUseYnAndPositionOrderBySortOrderAsc("Y", "MID");
    }

    // 전체
    public List<Banner> getAllBanners() {
        return bannerRepository.findAll();
    }

    public Banner createBanner(Banner banner) {
        return bannerRepository.save(banner);
    }

    public Banner updateBanner(Long id, Banner updated) {
        Banner banner = bannerRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("배너를 찾을 수 없습니다."));

        banner.setTitle(updated.getTitle());
        banner.setImageUrl(updated.getImageUrl());
        banner.setLinkUrl(updated.getLinkUrl());
        banner.setSortOrder(updated.getSortOrder());
        banner.setUseYn(updated.getUseYn());
        banner.setPosition(updated.getPosition());
        banner.setMainText(updated.getMainText());
        banner.setSubText(updated.getSubText());

        return bannerRepository.save(banner);
    }

    public void deleteBanner(Long id) {
        bannerRepository.deleteById(id);
    }
}

